pyof/utils.py

```diff
@@ -0,0 +1,37 @@
+"""General Unpack utils for python-openflow."""
+import pyof.v0x01.common.header
+import pyof.v0x01.common.utils
+import pyof.v0x04.common.header
+import pyof.v0x04.common.utils
+from pyof.foundation.exceptions import UnpackException
+
+pyof_version_libs = {0x01: pyof.v0x01,
+                     0x04: pyof.v0x04}
+
+
+def unpack(packet):
+    """Unpack the OpenFlow Packet and returns a message."""
+    try:
+        version = packet[0]
+    except IndexError:
+        raise UnpackException('null packet')
+
+    try:
+        pyof_lib = pyof_version_libs[version]
+    except KeyError:
+        raise UnpackException('Version not supported')
+
+    try:
+        header = pyof_lib.common.header.Header()
+        header.unpack(packet[:8])
```

Comment:
@erickvermot the header will always have length 8 [just to make sure that on OF 1.3.5 and newer they still keep this behavior]?
Or should we use `pyof_lib.common.header.Header.get_size()` instead of `8`?

If my comment make any sense, then we should "apply" it below also.